File: CustomerOrder/CustomerOrder.Service/Services/Client/ClientService.cs
using CustomerOrder.DTO;
using System;
using System.Linq;


namespace CustomerOrder.Service.Services.Client
{
    public class ClientService : BaseService
    {
        public DTO.ClientListDto GetClients(string filter, int skip, int take)
        {
            bool filterClient = true;
            if (string.IsNullOrEmpty(filter))
                filterClient = false;
            var qry = DbContext.client.Where(x =>

                filterClient ? x.email.Contains(filter) : true

            ).OrderBy(x => x.idClient);

            var count = qry.Count();
            var result = qry.Skip(skip).Take(take).Select(x => new DTO.ClientResponseDto()
            {
                ID_Client = x.idClient,
                Email = x.email,
                Nume = x.nume,
                Adresa = x.adresa,
            });
            return new DTO.ClientListDto()
            {
                ClientList = result,
                Count = count
            };
        }
        //public ClientResponseDto GetClient( int id)
        //{
        //    bool filterClient = true;
        //    if (id == 0)
        //    {
        //        filterClient = false;
        //    }
        //    //var qry = DbContext.client.Where(x => x.idClient.Equals(id) 
        //    var qry = DbContext.client.Where(x =>

        //        filterClient ? x.idClient.Equals(id) : true


        //    ).OrderBy(x => x.idClient);


        //    var result = qry.Select(x => new ClientResponseDto()
        //    {
        //        ID_Client = x.idClient,
        //        Email = x.email,
        //        Nume = x.nume,
        //        Adresa = x.adresa,
        //    });
        //    var i = result.ToList()[0];
        //    Console.WriteLine(result.ToList().GetRange(0, 1));
        //    return i;
        //}
        //public IEnumerable<ClientResponseDto> GetClients()
        //{
        //    var rsp = DbContext.client.Select(x => new ClientResponseDto()
        //    {
        //        ClientId= x.idClient,
        //        Email=x.email,
        //        Nume = x.nume,
        //        Adresa=x.adresa,
        //    });
        //    return rsp.ToList();
        //}

        public String AddClient(AddClientDto addClientDto)
        {
            Console.WriteLine(addClientDto);
            DAL.client client = new
                DAL.client();
            client.adresa = addClientDto.Adresa;
            client.nume = addClientDto.Nume;
            client.email = addClientDto.Email;

            var rsp = DbContext.client.Add(client);
            DbContext.SaveChanges();
            return "added";
        }
        public String EditClient(int id, EditClientDto editClientDto)
        {

            var client = DbContext.client.First(a => a.idClient == id);

            client.adresa = editClientDto.Adresa;
            client.nume = editClientDto.Nume;
            client.email = editClientDto.Email;

            DbContext.SaveChanges();
            return "updated";
        }
        public String DeleteClient(int id)
        {
            DbContext.client.Remove(DbContext.client.Find(id));


            DbContext.SaveChanges();
            return "deleted";
        }

    }
}

File: AspNetAngularClient/client/src/app/service/client.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Router} from '@angular/router';
import { ClientListDTO } from "./../shared/DTOs/clientListDTO";
import { ClientDTO } from "./../shared/DTOs/ClientDTO";

@Injectable({
  providedIn: 'root'
})
export class ClientService {
  base: string = 'http://localhost:60097/api/client/';

  constructor(private http: HttpClient, private router: Router) {
  }

  getClients(filter: string, skip: number, take: number) {
    return this.http.get<ClientListDTO>(this.base + 'getClients?filter=' + filter + '&skip=' + skip + '&take=' + take).toPromise();
  }

  getClient(id: number) {

    return this.http.get<ClientDTO>(this.base + 'getClient?id=' + id).toPromise();

    //  .map(clients => clients.find(client => client.ID_Client === id));
  }

  addClient(nume: string, email: string, adresa: string) {
    //   let client={ Nume:nume,Email:email,Adresa:adresa } as Client;
    const clientdto = {Nume: nume, Email: email, Adresa: adresa} as ClientDTO;
    return this.http.post(this.base + 'AddClient', clientdto).toPromise();

  }

  updateClient(id: number, nume: string, email: string, adresa: string) {

    const clientdto = {Nume: nume, Email: email, Adresa: adresa} as ClientDTO;

    // const url = `${this.base}/${id}`;
    return this.http.put(this.base + 'EditClient/' + id, clientdto).toPromise();
  }

  deleteClient(id: number) {

    return this.http.delete(this.base + 'DeleteClient/' + id).toPromise();
  }
}
